sklearn/cross_decomposition/_pls.py

```diff
@@ -722,6 +722,13 @@ def fit(self, X, y=None, Y=None):
         self.y_scores_ = self._y_scores
         return self
 
+    def __sklearn_tags__(self):
+        tags = super().__sklearn_tags__()
+        tags._xfail_checks.update(
+            {"check_non_transformer_estimators_n_iter": "Test doesn't apply."}
+        )
+        return tags
+
```

Comment:
If in the `check_non_transformer_estimators_n_iter` function we change

```python
assert np.all(estimator.n_iter_ >= 1)
```

to

```python
assert np.all(np.asarray(estimator.n_iter_) >= 1)
```

I suppose we don't need to `xfail` here (and for `PLSCanonical` and `CCA` as well)?